lib/github.com/diku-dk/containers/setops.fut

```diff
@@ -0,0 +1,141 @@
+import "../sorts/radix_sort"
+import "opt"
+
+-- | Set-like operations on arrays of values interpreted as sets
+
+local
+-- | The setops module type.  This module type is declared `local`, which means
+-- that it may not be referenced directly by name from client code.  This
+-- limitation makes it possible for the interface to be enriched by new members
+-- in future minor versions.
+
+module type setops = {
+  -- | [union a b] returns the set of elements that appear in both a and b with
+  -- dublicates removed. Work: O(n+m), span: O(1).
+  val union               [m][n] : [m]i64 -> [n]i64 -> ?[k].[k]i64
+
+  -- | [intersect a b] returns those elements in a that also appears in b with
+  -- dublicates removed. Work: O(n+m), span: O(1).
+  val intersect           [m][n] : [m]i64 -> [n]i64 -> ?[k].[k]i64
+
+  -- | [diff a b] returns a with dublicates and elements from b removed. Work:
+  -- O(n+m), span: O(1).
+  val diff                [m][n] : [m]i64 -> [n]i64 -> ?[k].[k]i64
+
+  -- | [elimdubs a] returns a with dublicates removed. Work: O(n), span: O(1).
+  val elimdubs               [m] : [m]i64 -> ?[k].[k]i64
```

Comment:
Typo: it is *duplicate*, and presumably `elimdups`.

Reply:
Nice catch! ;)